core/src/toga/handlers.py

```diff
@@ -215,6 +216,7 @@ def __init__(self, on_result: OnResultT | None = None) -> None:
                 "Synchronous `on_result` handlers have been deprecated; "
                 "use `await` on the asynchronous result",
                 DeprecationWarning,
+                stacklevel=2,
```

Comment:
As above, this is a pre-existing problem, but brackets around the multi-line string would be preferable.